sdk/python/kfp/compiler/_data_passing_rewriter.py

```diff
@@ -381,3 +383,337 @@ def deconstruct_single_placeholder(s: str) -> List[str]:
+            # to the input parameter per the implementation in _components.
+            # The only thing yet to be reconciled is the file name.
+
+            def reconcile_filename(
+                command_lines: List[str]) -> List[str]:
+                new_command_lines = []
+                for cmd in command_lines:
+                    matched = re.match(
+                        r'.*/{{workflow\.uid}}/{{inputs\.parameters\.'
+                        + input_name + r'}}/(?P<filename>.*)', cmd)
+                    if matched:
+                        new_command_lines.append(
+                            cmd[:-len(matched.group('filename'))] +
+                            output_to_filename[artifact_input['name']])
+                    else:
+                        new_command_lines.append(cmd)
+                return new_command_lines
+
+            if container_template['container'].get('args'):
+                container_template['container']['args'] = reconcile_filename(
+                    container_template['container']['args'])
+            if container_template['container'].get('command'):
+                container_template['container']['command'] = reconcile_filename(
+                    container_template['container']['command'])
+        else:
+            new_artifact_inputs.append(artifact_input)
+
+    container_template['inputs']['artifacts'] = new_artifact_inputs
+    container_template['inputs']['parameters'] = parameter_inputs
+
+
+def _refactor_dag_inputs(
+    dag_template: Dict[str, Any],
+    refactored_inputs: Dict[Tuple[str, str], str]
+) -> None:
+    """Refactors the inputs of the DAG template
+
+    Args:
+        dag_template: The DAG template structure.
+        refactored_inputs: The mapping of template and input names to be
+            refactored, to its new name.
+    """
+    # One hacky way to do the refactoring is by looking at the name of the
+    # artifact argument. If the name appears in the refactored_inputs mapping,
+    # this should be changed to a parameter input regardless of the template
+    # name.
+    # The correctness of this approach is ensured by the data passing rewriting
+    # process that changed the artifact inputs' name to be
+    # '{{output_template}}-{{output_name}}', which is consistent across all
+    # templates.
+    artifact_to_new_name = {k[1] : v for k, v in refactored_inputs.items()}
+    if not dag_template.get('inputs', {}).get('artifacts'):
+        return
+
+    parameter_inputs = dag_template['inputs'].get('parameters') or []
+    new_artifact_inputs = []
+    for input_artifact in dag_template['inputs']['artifacts']:
+        if input_artifact['name'] in artifact_to_new_name:
+            parameter_inputs.append(
+                {'name': artifact_to_new_name[input_artifact['name']]})
+            refactored_inputs[(dag_template['name'],
+                               input_artifact['name'])] = artifact_to_new_name[
```

Comment:
> `refactored_inputs[...] = artifact_to_new_name[...]`

Is this update needed? Aren't the values the same?

Reply:
When this function is called, refactored_inputs stores only the refactored inputs artifacts from container template, but not DAG templates. This statement adds new entries to the mapping